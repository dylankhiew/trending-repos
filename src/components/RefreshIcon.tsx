import * as React from 'react';
import { Image, StyleSheet, View } from 'react-native';

import { IMAGES } from '../../assets/IMAGES';

export default function RefreshIcon(): JSX.Element {
  return (
    <View style={styles.refreshIconContainer}>
      <Image source={IMAGES.ICON.REFRESH} style={styles.refreshIcon} />
    </View>
  );
}

const styles = StyleSheet.create({
  refreshIconContainer: {
    position: 'absolute',
    top: 30,
    height: 40,
    width: 40,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
    borderRadius: 40,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 5,
    },
    shadowOpacity: 0.44,
    shadowRadius: 5,
    elevation: 16,
    zIndex: 10,
  },
  refreshIcon: {
    height: 25,
    width: 25,
  },
});
