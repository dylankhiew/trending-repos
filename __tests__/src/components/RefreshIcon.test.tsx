import React from "react";
import renderer from "react-test-renderer";

import RefreshIcon from "../../../src/components/RefreshIcon";

it("should render correctly", () => {
  const tree = renderer.create(<RefreshIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});
