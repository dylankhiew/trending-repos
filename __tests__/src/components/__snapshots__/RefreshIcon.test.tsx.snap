// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 40,
      "elevation": 16,
      "height": 40,
      "justifyContent": "center",
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.44,
      "shadowRadius": 5,
      "top": 30,
      "width": 40,
      "zIndex": 10,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 25,
        "width": 25,
      }
    }
  />
</View>
`;
